### ISSUE TITLE
Docker v1.23-latest armv8 image is built for x86_64

### ISSUE THREAD

**Issue by milkcocoa** [envoyproxy/envoy#22261]
When I run the v1.23-latest image with my M1 MacBook Pro, I get the following error.

```
qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory
```

Installing `file` command, and I investigate the `envoy` command with it, the binary is built for x86-64 like the following.

```
$ file /usr/local/bin/envoy
/usr/local/bin/envoy: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=ce49c7f65668a22b80d1e83c35d170741bb8d46a, stripped
```

`uname` command shows the image itself is the armv8 architecture.

```
$ uname -a
Linux b2dff1cc33b1 5.10.104-linuxkit #1 SMP PREEMPT Thu Mar 17 17:05:54 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
```

**Comment by phlax** [comment]
@milkcocoa does doing the same thing with v1.22-latest work ?

**Comment by phlax** [comment]
cool, just wanted to see if it was a regression before tracking down - ill take a look

**Comment by milkcocoa** [comment]
v1.23.0, and distroless v1.23-latest, and v1.23.0 have the same behavior.

**Comment by phlax** [comment]
yep problem is caused by this https://github.com/docker/buildx/issues/510 bug in docker buildx

i added a default because it was breaking docs - there is a workaround in that ticket - i will PR a fix asap

**Comment by milkcocoa** [comment]
Great, thanks for the quick response, and investigation.

**Comment by phlax** [comment]
Fixed by #22444 i will close this once the 1.23.1 is released

**Comment by phlax** [comment]
fixed in version 1.23.1
